### PR TITLE
Remove links to dead orgs / meetups; fix some broken links

### PR DESCRIPTION
**Pull request by srowen**

### CODE REVIEW

File: developer-tools.md [hunk 1]
```diff
@@ -463,25 +463,16 @@ in the Eclipse install directory. Increase the following setting as needed:
...
-Spark developers access to the bleeding-edge of Spark master or the most recent fixes not yet 
-incorporated into a maintenance release. These should only be used by Spark developers, as they 
-may have bugs and have not undergone the same level of testing as releases. Spark nightly packages 
-are available at:
-
-- Latest master build: <a href="https://people.apache.org/~pwendell/spark-nightly/spark-master-bin/latest">https://people.apache.org/~pwendell/spark-nightly/spark-master-bin/latest</a>
-- All nightly builds: <a href="https://people.apache.org/~pwendell/spark-nightly/">https://people.apache.org/~pwendell/spark-nightly/</a>
-
-Spark also publishes SNAPSHOT releases of its Maven artifacts for both master and maintenance 
+Spark publishes SNAPSHOT releases of its Maven artifacts for both master and maintenance
```
Comment (srowen) [inline developer-tools.md:466]: We don't publish nightly builds anymore.

File: powered-by.md [hunk 2]
```diff
@@ -47,16 +47,13 @@ initially launched Spark
 - <a href="http://alluxio.com/">Alluxio</a>
   - Alluxio, formerly Tachyon, is the world's first system that unifies disparate storage systems
   at memory speed. 
-- <a href="http://alpinenow.com/">Alpine Data Labs</a>
```
Comment (srowen) [inline powered-by.md:50]: The removed orgs don't exist anymore

**Review by HyukjinKwon — APPROVED** [review]

File: developer-tools.md (hunk 1 above)
Comment (felixcheung) [reply developer-tools.md:466]: and maybe there's no point to publish SNAPSHOT artifacts? ...

File: powered-by.md [hunk 3]
```diff
@@ -170,9 +160,8 @@ across all screens
   - PanTera is a tool for exploring large datasets. It uses Spark to create XY and geographic 
   scatterplots from millions to billions of datapoints.
   - Components we are using: Spark Core (Scala API), Spark SQL, and GraphX
-- <a href="http://www.peerialism.com">Peerialism</a>
 - <a href="http://www.planbmedia.com">PlanBMedia</a>
-- <a href="http://prediction.io/">PredicitionIo</a>
+- <a href="http://predictionio.apache.org/index.html/">Apache PredicitionIo</a>
```
Comment (felixcheung) [inline powered-by.md]: `Apache PredictionIO`

**Review by felixcheung — COMMENTED** [review]
thanks!

File: powered-by.md (hunk 3 above)
Comment (srowen) [reply powered-by.md]: There may not be much point, yeah, in SNAPSHOT artifacts. I don't know how to shut them off -- @shaneknapp is that just a Jenkins job? The argument for turning it off is that it has been somewhat controversial in the past to provide any binaries from the project that aren't PMC releases. It's allowed if heavily signposted. But, maybe little point in dumping 1GB of artifacts per night into the repo.

Comment (shaneknapp) [reply powered-by.md]: we are currently not publishing any snapshot builds except for docs...  i can easily shut these builds off.

Comment (srowen) [reply powered-by.md]: The docs build probably isn't useful and just filling up dist, though that's not what I was referencing here. It's: https://repository.apache.org/snapshots/org/apache/spark and while that seems to down now, I did see recent snapshots of the artifacts. However @gatorsmile and others note that they may be useful, so let's keep whatever it is.

Comment (shaneknapp) [reply powered-by.md]: kk

**Pull request closed without merging**